docs_app/content/guide/best-practices.md

```diff
@@ -0,0 +1,100 @@
+
+    fromEvent(document.querySelector('input'), 'input').subscribe(event => {
+        ajax(`http://my-api/search?query=${event.target.value}`).subscribe(response => {
+            // handle the response
+        });
+    });
+
+Do this:
+
+    fromEvent(document.querySelector('input'), 'input').pipe(
+        switchMap(event => ajax(`http://my-api/search?query=${event.target.value}`))
+    ).subscribe(response => {
+        // handle the response
+    }); 
+
+
+### In general, heavy logic when subscribing is discouraged
+
+Here is a rule of thumb: 
+
+- A `subscribe` that has an anonymous function with many lines of code is bad;
+- A `subscribe` that has a named function as an argument is acceptable if nothing else can be done to simplify it using operators;
+- A `subscribe` that has an anonymous function with a single line of code is good;
+- A `subscribe` that has no arguments is great (though rare)
+- No `subscribe` is ideal
+
+## Avoiding memory leaks
+
+The way RxJS operates creates a way for memory leaks to appear in applications that use it. Usually it happens when a consumer subscribes to an `Observable` and the `Observable` itself never terminates naturally (for example, an `Observable` created by `interval`, which emits values in equal periods of time), and no one unsubscribes from it or terminates it manually. The simplest way to do that is directly unsubscribe when the subscription is no longer necessary:
+
+    const seconds$ = interval(1000);
+    const subscription = seconds$.subscribe(console.log)
+
+    // for example, unsubscribe when user hits a button
+
+    document.querySelector('button').addEventListener('click, () => subscription.unsubscribe())
+
+But this approach is imperative and does not convey explicitly when the subscription will terminate; if the two parts of code are separate from each other, just looking at the source `Observable` won't allow us to understand for just how long will the `Observable` sequence run. So a better approach is to use `takeUntil`:
+
+    interval(1000).pipe(
+        takeUntil(notifier$),
+    ).subscribe(console.log)
+
+In this example just looking at the `Observable` is enough to know exactly when it will terminate: when the `notifier$` `Observable` emits. Also, this is the widely accepted approach in the Angular community.
+
+### Be careful to avoid takeUntil leaks
+
+When `takeUntil` is not used correctly, it can lead to memory leaks too. As a rule of thumb, always place `takeUntil` as the last operator in the `pipe`. 
```

Comment:
can you elaborate on this a little bit to make it easier to understand for newcomers